avro-fastserde/src/main/java/com/linkedin/avro/fastserde/SchemaAssistant.java

```diff
@@ -25,20 +25,32 @@
 
 
 public class SchemaAssistant {
-  // The folllowing constants are not available in avro-1.4
+  // The following constants are not available in avro-1.4
   public static final String CLASS_PROP = "java-class";
   public static final String KEY_CLASS_PROP = "java-key-class";
 
   private final JCodeModel codeModel;
   private final boolean useGenericTypes;
   private Set<Class<? extends Exception>> exceptionsFromStringable;
+  /**
+   * This is used to collect all the fully qualified classes being used by the
+   * generated class.
+   * The purpose of this bookkeeping is that in some framework, those classes
+   * couldn't be found during compilation time, so we need to manually find
+   * the libraries, which define those classes, and put them in the compile classpath.
+   */
+  private final Set<String> fullyQualifiedClassNameSet = new HashSet<>();
 
   public SchemaAssistant(JCodeModel codeModel, boolean useGenericTypes) {
     this.codeModel = codeModel;
     this.useGenericTypes = useGenericTypes;
     this.exceptionsFromStringable = new HashSet<>();
   }
 
+  protected Set<String> getUsedFullyQualifiedClassNameSet() {
```

Comment:
Doesn't seem really "schema related". Should we move this functionality to `Utils` instead?

Reply:
This set is used to capture all the used classes during code generation (SchemaAssistant#classFromSchema is the function to extract all the used classes), and I think this is the right place to get this piece of info.